### PR TITLE
Add HIL interfaces for DMA and Caliptra Mailbox

### PR DESCRIPTION
**Pull request by mlvisaya**
Define HIL interfaces for Caliptra Mailbox and DMA.

This is based on the mailbox functionality defined in :

https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#mailbox

and the DMA operation in:

https://github.com/chipsalliance/caliptra-ss/blob/main/docs/Caliptra%202.0%20Subsystem%20Specification%201.pdf

### CODE REVIEW

File: runtime/mailbox/src/hil.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    ///
+    /// Returns:
+    /// - `Ok(response_length)` where `response_length` is the number of bytes read.
+    /// - `Err(ErrorCode)` if there was an issue reading the response.
+    fn read_response(&self, buffer: &mut [u8]) -> Result<usize, ErrorCode>;
+
+    /// Check the status of the mailbox after a command execution.
+    ///
+    /// Returns:
+    /// - `Ok(status_code)` where `status_code` represents the current mailbox status.
```
Comment (parvathib) [inline runtime/mailbox/src/hil.rs:51]: nit: Should we write `MailboxStatus` instead of `statis_code` for clarity?

**Review by parvathib — APPROVED** [review]

File: runtime/mailbox/src/hil.rs (hunk 1 above)
Comment (mlvisaya) [reply runtime/mailbox/src/hil.rs:51]: that's a good idea. i'll change it to MailboxStatus.

File: runtime/dma/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+edition.workspace = true
+
+[dependencies]
+kernel = { git = "https://github.com/tock/tock.git", rev = "b128ae817b86706c8c4e39d27fae5c54b98659f1" }
+
+[target.'cfg(target_arch = "riscv32")'.dependencies]
+capsules-core = { git = "https://github.com/tock/tock.git", rev = "b128ae817b86706c8c4e39d27fae5c54b98659f1" }
+registers-generated.workspace = true
+romtime.workspace = true
+tock-registers.workspace = true
```
Comment (swenson) [inline runtime/dma/Cargo.toml:16]: I don't think these are needed yet. (Same for mailbox.)
```suggestion
```

File: runtime/dma/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// Licensed under the Apache-2.0 license.
+
+#![cfg_attr(target_arch = "riscv32", no_std)]
+
+#[cfg(target_arch = "riscv32")]
```
Comment (swenson) [inline runtime/dma/src/lib.rs:5]: There's nothing RISC-V-specific about the HIL. It makes the build happier if we include it for all targets. (Same for mailbox.)

```suggestion
```

**Review by swenson — APPROVED** [review]
LGTM, but you can remove the RISC-V config stuff for the HILs at least.

**Pull request merged**